ProcessLib/BoundaryCondition/PhaseFieldIrreversibleDamageOracleBoundaryCondition.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "PhaseFieldIrreversibleDamageOracleBoundaryCondition.h"
+
+#include <algorithm>
+#include <logog/include/logog.hpp>
+#include <vector>
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+namespace ProcessLib
+{
+void PhaseFieldIrreversibleDamageOracleBoundaryCondition::getEssentialBCValues(
+    const double /*t*/, NumLib::IndexValueVector<GlobalIndexType>& bc_values) const
+{
+    SpatialPosition pos;
+
+    bc_values.ids.clear();
+    bc_values.values.clear();
+
+    // convert mesh node ids to global index for the given component
+    bc_values.ids.reserve(bc_values.ids.size() + _bc_values.ids.size());
+    bc_values.values.reserve(bc_values.values.size() +
+                             _bc_values.values.size());
+
+    std::copy(_bc_values.ids.begin(), _bc_values.ids.end(),
+              std::back_inserter(bc_values.ids));
+    std::copy(_bc_values.values.begin(), _bc_values.values.end(),
+              std::back_inserter(bc_values.values));
+}
+
+// update new values and corresponding indices.
+void PhaseFieldIrreversibleDamageOracleBoundaryCondition::preTimestep(
+    const double /*t*/, const GlobalVector& x)
+{
+    double irreversibleDamage = 0.05;
```

Comment:
A comment is needed for the number of 0.05.

Reply:
comments were added.